students/barb_matthews/lesson3/mailroom_one.py

```diff
@@ -0,0 +1,129 @@
+                    what = input("What donation amount to add?\n")
+                    huge_list[found][1].append(float(what))
+                    #print(huge_list[found])
+                    print("\n" * 100, "Email: Thank you, %s, for your generous donation of $%.2f. "
+                                     "We appreciate your support.\n" % (name, float(what)))
+                    time.sleep(3)
+                    break
+
+                else:
+                    donor_record = [name]
+                    huge_list.append(donor_record)      ## If new donor, add it to end of list
+                    found = len(huge_list) - 1
+
+                    ## prompt for donation amount, add it to name record in the index
+                    what = input("What donation amount?\n")
+                    huge_list[found].append([float(what)])
+                    #print(huge_list)
+                    print("\n" * 100, "Email: Thank you, %s for your generous donation of $%.2f. "
+                                  "We appreciate your support.\n" % (name, float(what)))
+                    time.sleep(3)
+                    break
+
+        elif choice.lower() == "list":
+            print("The current names are: ")
+            ## enumerate through the index and print names
+            for i, item in enumerate(huge_list):
+                print(item[0])
+            print('\n')
+
+        elif choice.lower() == "q":
+            break
+        else:
+            print("Try again.")
+    return
+
+def report():
+    """Prints a report to the screen of donors and amounts"""
+    print("\n" * 100, "\a{:<23} | {:<15} | {:<15} | {:>15}".format("Name", "Total Donated ($)",
+                                                                   "Number of Donations",
+                                                                   "Average Amount ($)"))
+    print("-" * 87)
+
+    ## Sort the list of donors in descending total donation amount order
+    s_mylist = sorted(huge_list, key=lambda record : sum(record[1]), reverse=True)
+
+    for person in s_mylist:
+        total = sum(person[1])
+        number = len(person[1])
+        average = total/number
+        print("{:<24} | {:>17,.2f} | {:>19} | {:>15,.2f}".format(person[0], total, number, average))
+
+    print("\n\n\n\n")
+    return
+
+def exit_menu():
+    """Exits out of the interactive menu"""
+    print("Ok, later")
+    sys.exit()
+
+def main():
+    """Prints the menu dispay to the screen and gets user input"""
+    while True:
+        response = input(prompt)  # continuously collect user selection
+        if response == "1":
+            thanks()
+        elif response == "2":
+            report()
+        elif response == "3":
+            exit_menu()
+        else:
+            print("\n" * 100)
+            print("Try again. Don't give up!\nWhat option?")
+
+# If not imported as a module, call the main function
+if __name__ == "__main__":
+    main()
```

Comment:
Much better!  Only one issue you'll want to look at.  Adding new donations to existing donors is working correctly, but adding new donors and donations isn't quite right.  Each new donation is creating a new line instead of adding to their total donations.